pwa-devdocs/src/technologies/basic-concepts/internationalization/index.md

```diff
@@ -0,0 +1,157 @@
+The language packages themselves are Magento extensions the application installs using Composer.
+
+For more information, see the Magento core topic: [Translations overview][].
+
+The tight coupling between core Magento's i18n feature and its frontend theme makes it difficult to use the same translation mechanisms in PWA Studio storefronts.
+Instead, PWA Studio provides its own i18n framework that follows a similar design as the one in core Magento.
+
+## How it works
+
+PWA Studio provides a context provider for translations called the [`LocaleProvider`][].
+This context provider contains translation data from dictionary files and supplies them to its child components.
+
+The [`react-intl`][] library powers the i18n framework in PWA Studio.
+It provides the [`IntlProvider`][] component to PWA Studio's `LocaleProvider`, which configures the component to use the storefront's translation data.
+This library also provides [`formatMessage()`][] and [`FormattedMessage`][] to localize text in child components.
+
+The following code samples produce the same results:
+
+```jsx
+return (
+    <p className={classes.text}>
+        <FormattedMessage id={LOREM_IPSUM} />
+    </p>
+)
+```
+
+```jsx
+const translatedText = formatMessage({ id: LOREM_IPSUM })
+
+return (
+    <p className={classes.text}>{translatedText}</p>
+)
+```
+
+The i18n framework uses the `id` parameter to look up the localized text from the dictionary files, which the framework supplies to the `LocaleProvider`.
+
+## Translation dictionaries
+
+Translation dictionary files contain key/value pairs for localized text.
+PWA Studio's i18n framework uses a similar dictionary approach for translation files as Magento, but
+instead of a CSV format, it uses JSON.
+
+Dictionary files must be inside an `i18n` directory and use their target locale for their filename.
+The proper format for the filename is: `<language id in lowercase>_<country id in uppercase>.json`.
+For example: `en_US.json`, `en_GB.json`, `fr_FR.json`.
+
+Example `en_US.json` content:
+
+```json
+{
+    "About Us": "About Us",
+    "Hi, {name}": "Hi, {name}",
+    "accountMenu.orderHistoryLink": "Order History",
+    "autocomplete.resultSummary": "{resultCount} items"
+}
+```
+
+### ID formats
+
+The JSON object's keys act as unique IDs for localized text.
+The example posted shows the different ways you can identify the localized text in a dictionary file.
+
+The first and second entry uses the original `en_US` locale text as the ID for the translated text.
+This is the same approach used in Magento.
+
+The third and fourth entry uses an alternative, dot notation approach for defining the ID for translated text.
+This is the main format used in PWA Studio components, and it helps identify the component and element where the text appears.
+
+Both approach have their pros and cons, and developers are free to choose which approach works for them when they develop their own components and storefronts.
```

Comment:
I would preface this by explaining our decision on using ID's `component.stringType`, it's documented in a number of comments in this issue: https://github.com/magento/pwa-studio/issues/669#issuecomment-696379447

Our messaging should be pro our ID's and then explain that whilst this is our best practice and how we intend to do our ID / message format that you can do this in a more Magento traditional way and won't be penalized.

Reply:
revised section